test/2-HomieNode.js

```diff
@@ -11,13 +11,13 @@ describe("Homie Node", function() {
   describe("Instantiation", function() {
 
     var testDevice = new HomieDevice('homie-device-test');
-    var testNode1 = testDevice.node('test-node-1', 'test-node');
+    var testNode1 = testDevice.node('test-node-1', 'friendly Name','test-node');
```

Comment:
If friendly name were added to the end of the list, would it make it compatible with earlier versions?

It's important to consider the impact of new features to existing implementations. Best case scenario, they can install the latest without any changes to their code - then migrate to new functionality as time permits.

If we introduce incompatibilities we'll have to bump the major version of this library. We may want or need to in order to support 3.x, but it's a huge win if we can bring existing users up to 3.x with little or no effort on their part.

Reply:
The friendly name in the homie 3.0.1 convention is not optional, so I decided to change this so it better match with the homie esp8266 API.

Sure I would prefer small changes too, but the specification changed alot since version 2. 
The most mqtt messages and node property definitions are completely different.